### PR TITLE
Fix TensorFlow's Squeeze and a new fusion for SoftMax from slim backend

### PR DESCRIPTION
**Pull request by dkurt**
### This pullrequest changes

resolves https://github.com/opencv/opencv/issues/13839

**Merge with extra**: https://github.com/opencv/opencv_extra/pull/595

### CODE REVIEW

File: modules/dnn/test/test_tf_importer.cpp [hunk 1]
```diff
@@ -675,6 +686,43 @@ TEST_P(Test_TensorFlow_layers, resize_bilinear)
...
+    int outShapes[][3] = {{3, 4, 2}, {1, 3, 2}, {1, 3, 4}, {1, 3, 4}};
+    int squeeze_dims[] = {0, 2, 3, -1};
+    for (int i = 0; i < 4; ++i)
+    {
+        std::string pbtxt =
+            "node { name: \"input\" op: \"Placeholder\""
+            "attr { key: \"data_format\" value { s: \"NHWC\" } } }"
+            "node { name: \"squeeze\" op: \"Squeeze\" input: \"input\""
+              "attr { key: \"squeeze_dims\" value { list { i:" + format("%d", squeeze_dims[i]) + "}}}}";
+        Net net = readNetFromTensorflow(0, 0, &pbtxt[0], pbtxt.size());
```
Comment (alalek) [inline modules/dnn/test/test_tf_importer.cpp]:
```.diff
-&pbtxt[0]
+pbtxt.c_str()
```

File: modules/dnn/test/test_tf_importer.cpp [hunk 2]
```diff
@@ -675,6 +686,43 @@ TEST_P(Test_TensorFlow_layers, resize_bilinear)
...
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE && target == DNN_TARGET_MYRIAD
+            && getInferenceEngineVPUType() == CV_DNN_INFERENCE_ENGINE_VPU_TYPE_MYRIAD_2
+    )
+        throw SkipTestException("Test is disabled for Myriad2");
+#endif
+    int inpShapes[][4] = {{1, 3, 4, 2}, {1, 3, 1, 2}, {1, 3, 4, 1}, {1, 3, 4, 1}};  // TensorFlow's shape (NHWC)
+    int outShapes[][3] = {{3, 4, 2}, {1, 3, 2}, {1, 3, 4}, {1, 3, 4}};
+    int squeeze_dims[] = {0, 2, 3, -1};
+    for (int i = 0; i < 4; ++i)
+    {
```
Comment (alalek) [inline modules/dnn/test/test_tf_importer.cpp]: You can try to use `SCOPED_TRACE(cv::format("i=%d", i));` for better error messages instead of custom message.

File: modules/dnn/test/test_tf_importer.cpp [hunk 3]
```diff
@@ -658,6 +658,17 @@ TEST_P(Test_TensorFlow_layers, softmax)
     runTensorFlowNet("slim_softmax");
 }
 
+TEST_P(Test_TensorFlow_layers, slim_softmax_v2)
+{
+#if defined(INF_ENGINE_RELEASE)
+    if (backend == DNN_BACKEND_INFERENCE_ENGINE && target == DNN_TARGET_MYRIAD &&
+        getInferenceEngineVPUType() == CV_DNN_INFERENCE_ENGINE_VPU_TYPE_MYRIAD_2
+    )
+        throw SkipTestException("Test is disabled for MyriadX");
```
Comment (alalek) [inline modules/dnn/test/test_tf_importer.cpp]: Myriad2?

Comment (dkurt) [reply modules/dnn/test/test_tf_importer.cpp]: Oh, sure, thank you!
